### ISSUE TITLE
is there any reproducibility problem?

### ISSUE THREAD

**Issue by herbertchen1** [Clarifai/few-shot-ctm#4]
It seems many people think it's hard to got the result.....

**Comment by hli2020** [comment]
I am checking it now and will respond to you asap.

**Comment by CoderHHX** [comment]
Yeah, I wish the author could release clean and reproducible codes, thanks!

**Comment by hli2020** [comment]
Hi, the core lies in the pre-trained model, which is trained on the meta training set. This is common practice as in [LEO](https://arxiv.org/pdf/1807.05960.pdf), [Qiao et al](https://arxiv.org/abs/1706.03466), for example. Unfortunately, the released version provides a wrong version (directly download the PyTorch model trained ImageNet); the performance in Table 4 (62.05% for 1-shot on miniImageNet) should be reproducible using the correct version of the pre-trained model. For now we can’t get access to the model we originally trained. Here is what we are gonna do within the couple days.

1. Clean up the codebase to make it more readable.
2. Train the pre-trained model and make it ready for the public.

Hopefully this could resolve some of your concerns. Stay tuned.

**Comment by VDeamoV** [comment]
I am trying to reproduce the Highest Score on mini-Imagenet (80.51%). Resnet18 is mentioned to use as the Backbone; however, I wonder if Relation Model, Reshaper, Concentrator, and Projector both change into BottleNeck, and the specific value for both m2 and m3 will be appreciated.

**Comment by VDeamoV** [comment]
Besides, I found the code in the `code`, it seems reshaper, concentrator, and projector both build with `Bottleneck`, so I wonder if this means that even backbone are 4 Conv, images from dataset still be processed by 4 Convs + 5 Bottlenecks with shotcuts (19 Conv Layers totally).

**Comment by munjalbharti** [comment]
Hi,
Can you please confirm if the configuration file 20way_1shot.yaml is configured for shallow or deeper network?
The structure is set to 'shallow' and dnet_out_c is 40.  To my understanding  dnet_out_c is the variable m2 in the paper. However in section 4.2.1 of paper you say m2 is 32 for 'shallow' network.
Can you please confirm if i need to change any configuration or use the configuration file as such?

Thanks,
Hoping for a reply :)

**Comment by hli2020** [comment]
Hi, a quick reply: please refer to the paper config. The 20way_1shot.yaml
is a bad example. I would fix it asap.

**Comment by EternalWang** [comment]
Could you release the 5-way-1-shot config file please? Thanks!

**Comment by DamonAtSjtu** [comment]
Hi, I want to know whether the current codes and configs are the correct version? I still cannot reproduce the result shown in the paper.

**Comment by haofengsiji** [comment]
the demo config using the backbone of 'shallow', I want to know can the backbone of 'ResNet19'  reproduce the paper? 
I always get the error of `ulrs error [Errno 111] Connection refused'`. I don't know why do I need to download something when I was training the model.

**Comment by haofengsiji** [comment]
I guess in the `core/config.py` has the true default set for 5 way 5 shot.
